lib/src/presentation/widget/similar_recipes_card.dart

```diff
@@ -0,0 +1,173 @@
+      decoration: BoxDecoration(
+        color: Colors.teal,
+        border: Border.all(
+          width: NumbersConstants.borderWith,
+        ),
+      ),
+      child: Padding(
+        padding: const EdgeInsets.only(
+          left: NumbersConstants.paddingXXXSmall,
+          top: NumbersConstants.paddingXXXSmall,
+          right: NumbersConstants.paddingXXXSmall,
+          bottom: NumbersConstants.paddingMedium,
+        ),
+        child: Column(
+          children: [
+            const Padding(
+              padding: EdgeInsets.only(
+                bottom: NumbersConstants.paddingXMedium,
+              ),
+              child: AutoSizeText(
+                textSimilarRecipes,
+                style: TextStyle(
+                  color: Colors.white,
+                  fontSize: textSimilarRecipesFontSize,
+                  fontWeight: FontWeight.w900,
+                  fontFamily: StringsConstants.textFontFamily,
+                ),
+                textAlign: TextAlign.center,
+                maxLines: NumbersConstants.autoSizeTextMaxLinesSecond,
+              ),
+            ),
+            SizedBox(
+              height: sizedBoxHeightListView,
+              child: ListView.builder(
+                itemCount: widget.similarRecipes.length,
+                itemBuilder: (context, index) {
+                  return Column(
+                    children: [
+                      Padding(
+                        padding: const EdgeInsets.only(
+                          bottom: NumbersConstants.paddingXXXSmall,
+                        ),
+                        child: InkWell(
+                          onTap: () async {
+                            await controller.getRecipeId(
+                              id: widget.similarRecipes[index].id,
+                            );
+                            Navigator.of(context).pushReplacementNamed(
+                              RoutesConstants.detailRoute,
+                              arguments: ParameterDetailPage(
+                                id: controller.state!.id,
+                                title: controller.state!.title,
+                                aggregateLikes:
+                                    controller.state!.aggregateLikes,
+                                instructions: controller.state!.instructions,
+                                image: controller.state!.image,
+                              ),
+                            );
+                          },
+                          child: SizedBox(
+                            width: sizeBoxWidthImageUrl,
+                            height: sizeBoxHeightImageUrl,
+                            child: CachedNetworkImage(
+                              imageUrl:
+                                  '${ApiServiceConstants.urlBaseImageSimilarRecipes}${widget.similarRecipes[index].id}${ApiServiceConstants.resolutionImageSimilarRecipes}.${widget.similarRecipes[index].imageType}',
```

Comment:
Instead to do this, you can create a variable like imageUrl and add fields to be replace like this:
`var imageUrl = 'https://spoonacular.com/recipeImages/<recipeIndex>-556x370.<imageType>'`

When you use this new variable, you are going to do something like this:
`imageUrl..replaceAll('<recipeIndex>', 6)..replaceAll('<imageType>', 'png')`

Reply:
Done